kube-core/src/response.rs

```diff
@@ -35,8 +33,69 @@ pub struct Status {
     pub details: Option<StatusDetails>,
 }
 
+impl Status {
+    /// Returns a successful `Status`
+    pub fn success() -> Self {
+        Status {
+            status: Some(StatusSummary::Success),
+            code: 0,
+            message: String::new(),
+            reason: String::new(),
+            details: None,
+        }
+    }
+
+    /// Returns an unsuccessful `Status`
+    pub fn failure(message: &str, reason: &str) -> Self {
```

Comment:
maybe we should split this up into a `failure` with `message` as the only param and have a `with_reason(reason)` function on `mut self` (or other way around if reason is the most important one - don't remember right now)

Reply:
actually, these two are very linked, it is probably a mistake to let people do one and not the other. ignore this.

while i am a bit uncomfortable with the double &str argument (which can be swapped around accidentally), there are probably smarter stuff we can do down the line to convert an `Error` into a `Status` by taking advantage of the `Display` and `Debug` traits for the error to fill these in a less error prone way.

Reply:
I can think of
```
fn from_error<E: Display>(err: &E, reason: &str) -> Self;
```
which uses e.to_string() as message (I don't see a way to infer reason from error, especially given guidelines that it shoud be short and PascalCase).

Reply:
Users could possibly do it in their app, if they did a convention like that if the Display was always PascalCase, and then they can take the Debug as the reason, but users would have to stay on top of their errors if so. We couldn't enforce that though, so I'd rather leave that blank and experiment in controller-rs with something like that.